Seg Fault on Error Log

Looks as if there is a nil pointer dereference exception when logging in NewClient in github.go

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x13080f7]

goroutine 1 [running]:
github.com/summerwind/actions-runner-controller/github.(*Config).NewClient(0xc0003cadc0, 0xc0000c0010, 0x2, 0x2)
	/workspace/github/github.go:50 +0x707
main.main()
	/workspace/main.go:85 +0x4eb
```

--- a/github/github.go
+++ b/github/github.go
@@ -16,7 +16,6 @@ import (
 
 // Config contains configuration for Github client
 type Config struct {
-	Log               logr.Logger
 	EnterpriseURL     string `split_words:"true"`
 	AppID             int64  `split_words:"true"`
 	AppInstallationID int64  `split_words:"true"`
@@ -46,7 +45,6 @@ func (c *Config) NewClient() (*Client, error) {
 	} else {
 		tr, err := ghinstallation.NewKeyFromFile(http.DefaultTransport, c.AppID, c.AppInstallationID, c.AppPrivateKey)
 		if err != nil {
-			c.Log.Error(err, "Authentication failed")
 			return nil, fmt.Errorf("authentication failed: %v", err)
 		}
 		httpClient = &http.Client{Transport: tr}
@@ -56,7 +54,6 @@ func (c *Config) NewClient() (*Client, error) {
 		var err error
 		client, err = github.NewEnterpriseClient(c.EnterpriseURL, c.EnterpriseURL, httpClient)
 		if err != nil {
-			c.Log.Error(err, "Enterprise client creation failed")
 			return nil, fmt.Errorf("enterprise client creation failed: %v", err)
 		}
 		githubBaseURL = fmt.Sprintf("%s://%s%s", client.BaseURL.Scheme, client.BaseURL.Host, strings.TrimSuffix(client.BaseURL.Path, "api/v3/"))

--- a/github/github.go
+++ b/github/github.go
@@ -9,7 +9,6 @@ import (
 	"time"
 
 	"github.com/bradleyfalzon/ghinstallation"
-	"github.com/go-logr/logr"
 	"github.com/google/go-github/v32/github"
 	"golang.org/x/oauth2"
 )

--- a/main.go
+++ b/main.go
@@ -82,15 +82,17 @@ func main() {
 	flag.DurationVar(&syncPeriod, "sync-period", 10*time.Minute, "Determines the minimum frequency at which K8s resources managed by this controller are reconciled. When you use autoscaling, set to a lower value like 10 minute, because this corresponds to the minimum time to react on demand change")
 	flag.Parse()
 
+	logger := zap.New(func(o *zap.Options) {
+		o.Development = true
+	})
+
 	ghClient, err = c.NewClient()
 	if err != nil {
 		fmt.Fprintln(os.Stderr, "Error: Client creation failed.", err)
 		os.Exit(1)
 	}
 
-	ctrl.SetLogger(zap.New(func(o *zap.Options) {
-		o.Development = true
-	}))
+	ctrl.SetLogger(logger)
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:             scheme,

